coderunners/scoring.py

```diff
@@ -0,0 +1,26 @@
+from abc import abstractmethod
+
+from models import Status, TestGroup
+
+
+class AbstractScorer:
+    @abstractmethod
+    def get_score(self, test_results: list[Status], test_scores: list[int]) -> float:
+        raise NotImplementedError()
+    
+class PerTestScorer(AbstractScorer):
+    def get_score(self, test_results: list[Status], test_scores: list[int]) -> float:
+        return sum(test_scores) / len(test_scores)
+
+class SubtaskScorer(AbstractScorer):
+    def __init__(self, test_groups: list[TestGroup]):
+        self.test_groups = test_groups
+        
+    def get_score(self, test_results: list[Status], test_scores: list[int]) -> float:
+        score = 0
+        test_results = test_results[:]
+        for test_group in self.test_groups:
+            oks = [int(test_result.status == Status.OK) for test_result in test_results[:test_group.count]]
+            score += sum(oks) * test_group.points_per_test + min(oks) * test_group.points
+            del test_results[:test_group.count]
```

Comment:
I think it would be better to loop through the test_results without modifying them. This can result in bugs in the future. Would be much better if we avoid having `del` of arrays in scoring. Especially `del` of arrays passed as arguments

Reply:
The list of copied before deletion starts to take place